### PR TITLE
Button: Tertiary & Button: Icon-Left/RIght

### LINKED ISSUE #84: Icon and link buttons
Need to complete the following to round out the button styles:
- Icon buttons
- Link buttons

icon-link-buttons

### PR DESCRIPTION
**Pull request by AndrwM**
Resolves: #84 

### Includes
- Merges `icon-button` with `button`
- Icon button modifier for either `left` or `right` icons (`.sage-btn.sage-btn--icon-left-<NAME>`)
- Tertiary button modifier (`.sage-btn.sage-btn--tertiary`)

### Preview
image

### 🍦 🍦 🍦 🍦 🍦

### CODE REVIEW

File: app/assets/stylesheets/sage/system/patterns/elements/_button.scss [hunk 1]
```diff
@@ -229,10 +266,35 @@ input[type="search"] {
   }
 }
 
-.sage-btn--link {
-  // TBD
+@mixin button-icon-generator($direction) {
```
Comment (philschanely) [inline app/assets/stylesheets/sage/system/patterns/elements/_button.scss:269]: 😍

File: app/assets/stylesheets/sage/system/patterns/elements/_button.scss [hunk 2]
```diff
@@ -229,10 +266,35 @@ input[type="search"] {
...
+  @if $direction == right {
+    $right: true;
+  } @else if $direction == left {
+    $right: false;
+  } @else {
+    @error "button-icon-generator() requires argument value of either `right` or `left`";
+  }
+
+  @each $icon-name, $icon-code in $sage-icons {
+    .sage-btn--icon-#{if($right, "right", "left")}-#{$icon-name} {
```
Comment (philschanely) [inline app/assets/stylesheets/sage/system/patterns/elements/_button.scss:281]: 🙌 for doing this in a combined fashion with one modifier versus separating them. This is very useful!!

**Review by philschanely — APPROVED** [review]
Marvelous and elegant!

**Review by teenwolfblitzer — APPROVED** [review]
👍 🔥 Great work! As we discussed yesterday, the only comment I'll add is about the classname list getting long here, but we'll revisit and refactor if necessary when the "default" is decided on.

**Pull request merged**